internal/common/voting/election.go

```diff
@@ -86,8 +86,114 @@ func (e *Election) CloseBallot() shared.ClientID {
 }
 
 func (e *Election) bordaCountResult() shared.ClientID {
-	// TODO implement Borda count winner selection method.
-	return e.pluralityResult()
+	// Implement Borda count winner selection method
+	//(may need to modify if methods design is changed)
+	var candidatesNumber int = 0
+	var islandsNumber int
+	var votesLayoutElect map[int][]int
+	votesSliceSquare := e.votes
+	candidatesNumber = len(e.islandsToVote)
+	islandsNumber = len(votesSliceSquare)
+
+	//Initialize votesLayoutMap
+	for i := 1; i < islandsNumber+1; i++ {
+		for j := 0; j < candidatesNumber; j++ {
+			votesLayoutElect[i] = append(votesLayoutElect[i], 0)
+		}
+	}
+	//Transfer e.votes to votesLayoutMap with type "int"
+	for i := 0; i < islandsNumber; i++ {
+		scoreInit := candidatesNumber + 1
+		for j := 0; j < candidatesNumber; j++ {
+			for k := 0; k < candidatesNumber; k++ {
+				if votesSliceSquare[i][j] == e.islandsToVote[k] {
```

Comment:
are there any checks to make sure voteSliceSquare is properly formed? If a client votes has duplicate entries then wouldn't this simply assign the lower score for that candidate.

i.e. if the input was {6,1,2,3,6,4}
the score for candidate 6 wouldn't be  7 it would be 3

Reply:
from my understanding, the input for borda count i.e.  {3,4,5,1,2,6} should be a preference list in which candidates are sorted by popularity, so a candidate should not appear twice in it
if this happens, it should be a problem with GetVoteForElection in baseclient which I think is not completed yet?